src/input/mod.rs

```diff
@@ -2651,6 +2651,10 @@ pub fn apply_libinput_settings(config: &niri_config::Input, device: &mut input::
 
         if let Some(method) = c.scroll_method {
             let _ = device.config_scroll_set_method(method.into());
+
+            if method == niri_config::ScrollMethod::OnButtonDown {
+                let _ = device.config_scroll_set_button(c.scroll_button);
+            }
```

Comment:
I guess these should be outside of the `if` in case the default scroll method for the device is already `OnButtonDown`?